File: docs/js/api/comments.js
"use strict";

import { BASE_URL, requestOptions } from "./common.js";

const commentsAPI = {

    get_comments: function () {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/comments`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },

    get_comment_by_id: function (commentId) {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/comments/${commentId}`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },

    get_comment_by_photo_id: function (photoId) {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/comments/photo/${photoId}`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => error);
            });
    },

    get_comment_num_by_photo_id: function (photoId) {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/comments/num/photo/${photoId}`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => error);
            });
    },

    create: function (formData) { //POST
        return new Promise(
            function (resolve, reject) {
                axios.post(`${BASE_URL}/comments`, formData, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
        });
    },

};

export { commentsAPI };

File: endpoints/comments.py
from silence.decorators import endpoint

###############################################################################

@endpoint(
    route="/comments",
    method="GET",
    sql="SELECT comments.commentId,comments.txt,comments.dateP,comments.userId,comments.photoId FROM comments JOIN photos ON (comments.photoId=photos.photoId) WHERE photos.visibility='Pública' ORDER BY comments.dateP DESC"
)
def get_comments():
    pass

###############################################################################

@endpoint(
    route="/comments/$commentId",
    method="GET",
    sql="SELECT * FROM Comments WHERE commentId = $commentId",
    description="Gets a comment by its id",
)
def get_comment_by_id():
    pass

###############################################################################

@endpoint(
    route="/comments/photo/$photoId",
    method="GET",
    sql="SELECT * FROM Comments WHERE photoId = $photoId",
    description="Gets all comments of a photo by photoId",
)
def get_comment_by_photo_id():
    pass

###############################################################################

@endpoint(
    route="/comments/num/photo/$photoId",
    method="GET",
    sql="SELECT COUNT(*) num FROM Comments WHERE photoId = $photoId",
    description="Gets the number of comments that have a photo with photoId=$photoId",
)
def get_comment_num_by_photo_id():
    pass

###############################################################################

@endpoint(
    route="/comments",
    method="POST",
    sql="INSERT INTO Comments (txt, userId, photoId) VALUES ($txt, $userId, $photoId)",
    description="Creates a new comment",
    auth_required=True,
)
def create(txt, userId, photoId):
    pass

###############################################################################